### DISCUSSION TITLE
Is it permitted to do Remote Requests for Service Monitoring with Git Actions?

### DISCUSSION THREAD

**Discussion by Aklakan** [community/community#26140] [Actions]
<p>Hi, I have a repo that contains around 500 URLs of remote data services which occasionally I want to check for online/offline status (and in the future possibly query these services for metrics) and publish the results as an RDF dataset.</p>
<p>I did a proof of concept setup that only checks 5 URLs and its works: <a href="https://github.com/SmartDataAnalytics/lodservatory/blob/master/latest-status.trig" rel="nofollow noopener">This dataset</a> with service status is generated by a git action. However, of course I want to avoid a ban e.g. due to e.g. a violation of the terms-of-service or recurring overuse of quotas.</p>
<p>So is the use of git actions for connecting recurrently checking remote URLs permitted? For example: once every day - if there are per-hour-quotas, then distributed over several hours.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/aklakan">@aklakan</a> ,</p>
<p>This is permitted. You can use github actions(eg: schedule event) to connect recurrently checking remote URLs.</p>
<p>AFAIK, since you’re monitoring 500URLs, what you should be aware is that the usage limit <a href="https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits" rel="nofollow noopener">here</a>(eg: 1000 API requests up…etc) and not exceeds the github hosted runner <a href="https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources" rel="nofollow noopener">storage</a>.</p>
<p>Hope it helps!</p>
